Gems/AzQtComponentsForPython/Code/CMakeLists.txt

```diff
@@ -0,0 +1,146 @@
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzQtComponents
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzCore/Platform/${PAL_PLATFORM_NAME} 
+        -I${QT_MODULE_DIR}/include
+        -I${QT_MODULE_DIR}/include/QtWidgets
+        -I${QT_MODULE_DIR}/include/QtCore
+        -I${QT_MODULE_DIR}/include/QtGui
+        -T${PYSIDE2_TYPESYSTEM_PATH}
+        -T${PYSIDE2_SHARE_PATH}
+        --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+    )
+
+    set(generated_sources_dependencies ${wrapped_header} ${typesystem_file})
+
+    add_custom_command(
+        OUTPUT ${WRAPPED_FILES}
+        COMMAND ${SHIBOKEN_GENERATOR_EXE_PATH} ${shiboken_options} ${wrapped_header} ${typesystem_file}
+        DEPENDS ${generated_sources_dependencies} 
+        COMMENT "Running generator for ${typesystem_file}."
+        VERBATIM
+    )
+
+    # Set up the project for building the code generated by shiboken.
+    ly_add_target(
+        NAME ${GEM_NAME} GEM_MODULE
+        NAMESPACE Gem
+        OUTPUT_NAME ${BINDINGS_LIBRARY}
+        FILES_CMAKE
+            wrapped_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                ${PYSIDE2_INCLUDE_DIR}
+                ${PYTHON_INCLUDE_DIR}
+                ${SHIBOKEN_INCLUDE_DIR}
+                ${PYSIDE2_INCLUDE_DIR}\\QtCore
+                ${PYSIDE2_INCLUDE_DIR}\\QtGui
+                ${PYSIDE2_INCLUDE_DIR}\\QtWidgets
+        BUILD_DEPENDENCIES
+            PUBLIC
+                3rdParty::pyside2
+                Qt5::Widgets
+                Qt5::Core
+                Qt5::Widgets
+                Qt5::Gui
+                ${AZQTCOMPONENTS_LIBRARY}
+                Gem::EditorPythonBindings.Static
+    )
+
+    ly_create_alias(NAME ${MODULE_NAME}.Builders NAMESPACE Gem TARGETS ${MODULE_NAME})
+    ly_create_alias(NAME ${MODULE_NAME}.Tools NAMESPACE Gem TARGETS ${MODULE_NAME})
+    ly_create_alias(NAME ${MODULE_NAME}.Clients NAMESPACE Gem TARGETS ${MODULE_NAME})
+
+    target_link_libraries( ${GEM_NAME} PRIVATE 
+        debug ${PYSIDE2_SHARED_LIBRARIES_D}
+        optimized ${PYSIDE2_SHARED_LIBRARIES}
+    )
+
+    target_link_libraries( ${GEM_NAME} PRIVATE 
+        debug ${SHIBOKEN_SHARED_LIBRARIES_D}
+        optimized ${SHIBOKEN_SHARED_LIBRARIES}
+    )
+
+    target_sources(${GEM_NAME} PRIVATE ${WRAPPED_FILES})
+
+    # Building the wrapper files will not work with a unity build.
+    set_target_properties(${GEM_NAME} PROPERTIES UNITY_BUILD OFF)
+
+    # Append _d to the module name in a debug build.
+    set_target_properties(${GEM_NAME} PROPERTIES DEBUG_POSTFIX "_d")
+
+    if (WIN32)
+        # Disable various warnings in shiboken generated wrapper code.
+        #"conditional expression is constant"
+        #"unreferenced formal parameter"
+        #"declaration of 'x' hides previous local declaration."
+        #"declaration of 'x' hides class member."
+        target_compile_options(${GEM_NAME} PRIVATE /wd4127 /wd4100 /wd4456 /wd4458)
+    endif()
```

Comment:
This should check the [PAL_PLATFORM_NAME](https://github.com/o3de/o3de/blob/development/cmake/Platform/Windows/PALDetection_windows.cmake#L9) opt instead of directly referencing WIN32.

WIN32 maybe defined by "other" platforms